index.go

```diff
@@ -289,28 +290,33 @@ func LoadIndex(path string) (*Index, error) {
 		}
 	}
 
-	var reader io.ReadCloser
 	fh, err := os.Open(idxpath)
 	if err != nil {
 		return nil, err
 	}
 	defer fh.Close()
-	reader = zstd.NewReader(fh)
-	defer reader.Close()
 
-	data, err := ioutil.ReadAll(reader)
+	reader := bufio.NewReader(fh)
+
+	firstLine, err := reader.ReadBytes('\n')
 	if err != nil {
 		return nil, err
 	}
-	index := Index{List: []IndexEntry{}}
-	yaml.Unmarshal(data, &index)
-
-	// Check index.Filepath == path
-	if index.Filepath != path {
-		return nil, ErrIndexPathMismatch
+	var index Index
+	err = json.Unmarshal(firstLine, &index)
+	if err != nil {
+		return nil, err
 	}
 
-	// Check file is not newer than index
+	//
+	// This check is counter productive, because it makes the indices tied to
+	// the file system they were created on: you cannot copy them to another
+	// location.
+	//
+	// if index.Filepath != path {
+	//	return nil, ErrIndexPathMismatch
+	// }
+
```

Comment:
I don't like this change - I think there should be some kind of explicit linkage between the index and the dataset. Using an incorrect index is likely not going to crash, but will produce erroneous results.

I agree that absolute paths might be overly strict. What about using a relative path like `./rir_clc_ipv_range.csv`, and making the path absolute at test time using the basepath of the index?

I'm okay if you want to leave it commented out for now, but add a FIXME comment so one of us can revisit it.

Reply:
How about adding a method to the Index class for checking at various levels of strictness?

Reply:
I can think of other options for handling errors that are non-fatal:

1. Return an additional parameter ([]string or []error, containing things like the mismatch above)
2. Add an additional field to the Index containing the warnings encountered during loading (do not serialize this field)
3. Use MD5 hash instead of pathname to ensure the file and index belong together.  This could be a bit slow, about 10s for a 10GB file, so we would probably make this an optional check, e.g. Index.VerifyMD5(path) after the index is loaded.
4. Use attributes other than the _absolute_ file path. The file name kind of works, except for cases where people just call their files "data" (relying on the path to discriminate). You could also use the uncompressed file size. This option is faster than MD5 but more fragile.

Let me know if you can think of any other